AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/hierarchical-state-machine-swiftTests/StateMachineTests.swift

```diff
@@ -76,23 +62,18 @@ class StateMachineTests: XCTestCase {
     /// receiving an event. `receive` is asynchronous, so we'd expect some of
     /// the current values to be out of sync if the state machine didn't
     /// properly serialize access.
-    func testConcurrentReceiveAndRead() {
+    func testConcurrentReceiveAndRead() async  {
         // Logging will significantly impact performance on these tight loops, so adjust expectation
         // timeout accordingly if you need to log
         let testMachine = CounterStateMachine.default()
         let increment = Counter.Event(id: "1", eventType: .increment)
-        var allExpectations = [XCTestExpectation]()
         for iteration in 1 ... 10 {
-            testMachine.send(increment)
-            let notified = expectation(description: "notified")
-            allExpectations.append(notified)
-            testMachine.getCurrentState {
-                XCTAssertEqual($0.value, iteration)
-                notified.fulfill()
-            }
+            await testMachine.send(increment)
+
+            let state = await testMachine.getCurrentState
```

Comment:
If we do adjust the `send` behavior to return the current state, we should retain this test: `machine.currentState` should still be actor-isolated and therefore async